tests/test_2way.py

```diff
@@ -0,0 +1,140 @@
+"""
+Bidirectional streaming and context API.
+"""
+
+import trio
+import tractor
+
+# from conftest import tractor_test
+
+# TODO: test endofchannel semantics / cancellation / error cases:
+# 3 possible outcomes:
+# - normal termination: far end relays a stop message with
+# final value as in async gen from ``return <val>``.
+
+# possible outcomes:
+# - normal termination: far end returns
+# - premature close: far end relays a stop message to tear down stream
```

Comment:
Near end sends a message, not far end, right?

Reply:
It should eventually be one of either, we'll need tests for both cases.

You should be able to close either side's `Context.open_stream()` and the other side shuts down gracefully i'm thinking?

Reply:
That makes sense to me. I figured you were laying out how it might happen from either side